src/main/java/org/mskcc/cbio/oncokb/service/UserService.java

```diff
@@ -701,17 +704,13 @@ public Optional<UserDTO> updateUserWithCompanyLicense(UserDTO userDTO, Company c
      */
     private void expireUserAccount(UserDTO userDTO) {
         userDTO.setActivated(false);
-        Optional<UserDTO> optionalUpdatedUserDTO = updateUser(userDTO);
-        if (optionalUpdatedUserDTO.isPresent()) {
-            UserDTO updatedUserDTO = optionalUpdatedUserDTO.get();
-            if (userHasUnactivatedTrial(updatedUserDTO)) {
-                clearTrialAccountInformation(updatedUserDTO);
-            }
-            List<Token> tokens = tokenService.findByUser(userMapper.userDTOToUser(userDTO));
-            tokens.forEach(token -> {
-                tokenService.expireToken(token);
-            });
+        if (userHasUnactivatedTrial(userDTO)) {
+            clearTrialAccountInformation(userDTO);
         }
+        List<Token> tokens = tokenService.findByUser(userMapper.userDTOToUser(userDTO));
+        tokens.forEach(token -> {
+            tokenService.expireToken(token);
+        });
```

Comment:
Why we want to make this change?

Reply:
Calling updateUser would result in an infinite loop because expireToken is called in updateUser. It is unnecessary because the user is already checked and updated before it is passed in to expireToken in the updateUser method

Reply:
But this method needs to inactivate the userDTO if it hasn't. Let's add a conditional check on activated status, it's still true, we need to set it to false and call updateUser.

Reply:
This method does still inactivate the userDTO before the lines I added. I will add the conditional check